### ISSUE TITLE
The CanICA loop could be made parallel

### ISSUE THREAD

**Issue by GaelVaroquaux** [nilearn/nilearn#235]
In CanICA, the fastica loop could be parallelized.

I have in mind the 'for rs in range(self.n_init)...' loop in the fit of CanICA.

One thing to keep in mind is that random_state need to be reseeded after a fork (elsewhere the seed is the same everywhere). Thus we can parallelize the loop only for scikit-learn > 0.13 (I suggest setting n_job to one else) and we need to use a pattern resembling https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/cluster/k_means_.py#L306
